sdk/core/azure-core/src/main/java/com/azure/core/util/tracing/Utils.java

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.tracing;
+
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+
+import static com.azure.core.util.tracing.Tracer.ENTITY_PATH_KEY;
+import static com.azure.core.util.tracing.Tracer.HOST_NAME_KEY;
+import static com.azure.core.util.tracing.Tracer.MESSAGE_ENQUEUED_TIME;
+
+final class Utils {
+
+    static final String SPAN_KIND_KEY = "span-kind";
+    static final String START_TIME_KEY = "span-start-time";
+
+    static final AutoCloseable NOOP_CLOSEABLE = () -> { };
+    static final ClientLogger LOGGER = new ClientLogger(Utils.class);
+
+    private Utils() {
+    }
+
+    @SuppressWarnings("deprecation")
+    static void addMessagingAttributes(StartSpanOptions spanOptions, Context context) {
+        String entityPath = getOrNull(context, ENTITY_PATH_KEY, String.class);
+        if (entityPath != null) {
+            spanOptions.setAttribute(ENTITY_PATH_KEY, entityPath);
+        }
+        String hostName = getOrNull(context, HOST_NAME_KEY, String.class);
+        if (hostName != null) {
+            spanOptions.setAttribute(HOST_NAME_KEY, hostName);
+        }
+        Long messageEnqueuedTime = getOrNull(context, MESSAGE_ENQUEUED_TIME, Long.class);
```

Comment:
This logic here applies when deprecated Tracer APIs are called. So we should not expect it to be called much after a few releases and eventually should be removed altogether. 


Beyond this API, I think it would be great to expose iterator on context  - this way it should be more flexible when multiple entries are requested. Like `getValues()`, but without Map and deduplication